var/spack/repos/builtin/packages/r/package.py

```diff
@@ -82,6 +82,8 @@ class R(Package):
     depends_on('pcre')
     depends_on('jdk')
 
+    patch('zlib.patch', when='@:3.3.1', level=0)
```

Comment:
Also, did you confirm that the patch properly applies to older versions like 3.1.2?